tools/testing/fetch_mailman.sh

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+
+FALCON_ROOT=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )/../.." &> /dev/null && pwd )
+MAILMAN_PATH=$FALCON_ROOT/.ecosystem/mailman
+
+# TODO(vytas): Detect the latest version from git tags or PyPi JSON.
+# NOTE(vytas): One approach: `curl -Ls https://pypi.org/pypi/mailman/json | jq -r .info.version`
+# MAILMAN_VERSION=3.3.5
+
+# Clean up in case we are running locally and not in CI
+rm -rf $MAILMAN_PATH
+
+mkdir -p .ecosystem
+git clone https://gitlab.com/mailman/mailman.git/ $MAILMAN_PATH
+
+# TODO(vytas): Enable version checking when a stable release's tests pass as-is.
+#   At the time of writing, the latest version tag (3.3.5) has some failing tests.
+cd $MAILMAN_PATH
+# git checkout tags/$MAILMAN_VERSION
+
+# NOTE(vytas): Patch tox.ini to introduce a new Falcon environment.
+cat <<EOT >> tox.ini
+
+[testenv:falcon]
+commands =
+    pip uninstall -y falcon
+    pip install $FALCON_ROOT
+    python -m nose2 -v {posargs}
```

Comment:
Instead of overriding the `commands`, you can add a new `commands_pre` section, which will install falcon. It might be better if in future we decide to change the actual `command` we use in CI.

https://tox.wiki/en/latest/config.html#conf-commands_pre

Reply:
Spot on. Done :heavy_check_mark: